pkg/generate/app/pipeline.go

```diff
@@ -222,7 +222,7 @@ func AddServices(objects Objects) Objects {
 				}
 			}
 			if len(svc.Spec.Ports) == 0 {
-				glog.Warningf("DeploymentConfig %q: Cannot create a service with no ports", t.Name)
+				glog.Warningf("A service will not be generated for DeploymentConfig %q because no exposed ports were detected. Use 'osc expose dc %q --port=[port] --generator=service/v1' to create a service.", t.Name, t.Name)
```

Comment:
it works with the quoted name, and it helps the shell interpret it as a single argument